elastic/kibana
# [APM][Otel] Add Otel client based on PoC data

Closes [#192115](https://github.com/elastic/kibana/issues/192115)
Closes [#192465](https://github.com/elastic/kibana/issues/192465)


## Summary

This PR adds synthrace client for Otel native data and a simple scenario. This is the first step of adding it and in the future it will include more metrics and use cases.

>[!NOTE]
> To run ES the command needs "xpack.otel_data.registry.enabled=true" flag
> `yarn es snapshot --license trial --E "xpack.otel_data.registry.enabled=true"`

## Next steps
- We currently have only `service_destination` in the metrics indices we can include the other types in the future
- After we have all the UI changes we can add more scenarios (also using the opentelemetry demo data and not only the e2e PoC example)

## Testing
- Run ES: 
```bash 
yarn es snapshot --license trial --E "xpack.otel_data.registry.enabled=true"
```
- Run Kibana:
```bash 
yarn start
```

>[!WARNING]
If the e2e PoC is used the first 2 steps should be skipped

- Run syntrace: 
```bash
node scripts/synthtrace otel_simple_trace.ts --clean
```
- Check indices in DevTools for the generated data: 
```bash 
GET *metrics-generic.otel*/_search

GET *traces-generic.otel*/_search

GET *logs-generic.otel*/_search
```
- Check in the APM UI (all the tabs) 
>[!WARNING]
 Currently the UI changes done in APM are not merged so some errors are expected)

https://github.com/user-attachments/assets/92f63610-82da-40f3-89bb-00be83c55377
